### ISSUE TITLE
renameKeysWith vs renameKeysBy

### ISSUE THREAD

**Issue by lgy87** [char0n/ramda-adjunct#653]
i note that the signature of renameKeysWith is `(a -> b) -> {a: *} -> {b: *}`
the first argument is unary，so is it better rename it to `renameKeysBy` according to lodash and ramda?

**Comment by char0n** [comment]
There was an issue in ramda what is the distinction between the *By and *With. I'll try to find it. Can you pls provide some formalization of the rules if you are aware of it ?

**Comment by lgy87** [comment]
```
// nope
R.startsWith
R.endsWith

// special
R.useWith
R.pickBy            first argument is binary
R.reduceBy       first arg is binary

// normal
// first argument is binary
R.differenceWith
R.dropRepeatsWith
R.groupWith
R.memoizeWith
R.mergeDeepWith
R.symmetricDifferenceWith
R.unionWith
R.uniqWith
R.zipWith

// first argument is unary
R.countBy
R.eqBy
R.groupBy
R.indexBy
R.maxBy
R.minBy
R.sortBy
R.uniqBy
```

as far as i know, all `*with`(except **startsWith**, **endsWith**) and `*By` obey the regularity in lodash.

please correct me if i am wrong.

**Comment by char0n** [comment]
Your argumentation seems valid. What we can do to create alias for `renameKeysWith` and call it `renameKeysBy`. Or we can extend the current implementatin and allow a value to be passed inside the function along with the key.

**Comment by lgy87** [comment]
I prefer the first solution to the second. but if `renameKeysWith`(take a binary callback) is also need, there will be a broken update. 😢

**Comment by char0n** [comment]
You're right. Second solution would be a breaking change. Callback would consume value as the first argument and key as second. I tend to like the second option, which extend the functionality of the function. @Undistraction, @guillaumearm, @srghma, @rockymadden opinions on this ?

**Comment by rockymadden** [comment]
First path with a caveat:

- A new binary version of `renameKeysWith` is made a la something like https://ramdajs.com/docs/#mapObjIndexed
- `renameKeysWith` is implemented with the above function, whatever it is called. It essentially is implemented ignoring the value parameter in its implementation.
- `renameKeysWith` has an alias/deprecation to `renameKeysBy` and signature remains the same.

**Comment by char0n** [comment]
@rockymadden thanks make sense to me. We can introduce new `renameObjKeysWith` and `renameObjKeysBy` and let `renameKeysWith` by an alias or `renameObjKeysBy`. What do you think ? //cc @lgy87

**Comment by lgy87** [comment]
@char0n Sounds good。

**Comment by char0n** [comment]
OK so decided then. I'll start executing this.

**Comment by char0n** [comment]
I'll address this next
